### PR TITLE
Handle multiple version of notebooks

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**: 
Jira: https://issues.redhat.com/browse/RHODS-788

**Analysis / Root cause**: 
UI does not support new format of notebook image versions.

**Solution Description**: 
Update the UI to use the new data format retrieved for images. Show the different versions for the notebooks as radio buttons which allow the user to select their desired version. Indicate which version is recommended and order the notebooks based on the `order` field.

**Screen shots / Gifs for design review**: 
image

**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

/cc @kywalker-rh

### CODE REVIEW

**Comment by kywalker-rh** [comment]
Is it possible to keep the second column spacing the same whether the first column is expanded or not? I.e. the spacing between Standard data science and SciPy would be the same as SciPy and Minimal Python, whether the first column in expanded or not?

File: jupyterhub_singleuser_profiles/ui/src/ImageForm/imageUtils.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ImageSoftwareType, ImageTagType } from '../utils/types';
+
+export const getNameVersionString = (software: ImageSoftwareType): string => {
+  const versionString = software?.version ? ` ${software?.version}` : '';
```
Comment (cdcabrera) [inline jupyterhub_singleuser_profiles/ui/src/ImageForm/imageUtils.ts:4]: comment/question... if version is always "truthy", or has a "V" in it, I'm good with this... any chance version will ever be `0`? 

probably don't need the `?` in ` ${software?.version}`, works regardless

File: jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageTagPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { ImageSoftwareType, ImageTagType } from '../utils/types';
+
+type ImageTagPopoverProps = {
+  tag: ImageTagType;
+};
+
+const getNameVersionString = (software: ImageSoftwareType): string => {
+  const versionString = software?.version ? ` ${software?.version}` : '';
+  return `${software.name}${versionString}`;
+};
```
Comment (cdcabrera) [inline jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageTagPopover.tsx:13]: question: repeated function from `imageUtils.ts`?

**Review by cdcabrera — APPROVED** [review]
Couple of questions... repeated code, nothing blocking

**Comment by mroman-redhat** [comment]
When testing on cluster, selecting out of two versions, the user configmap received this: `last_selected_image: '[object Object]:v0.0.14'`

**Comment by vpavlin** [comment]
It seems to be working fine for me
Screenshot from 2021-06-30 16-10-26

**Comment by vpavlin** [comment]
Ok, I have to change my statement - the UI looks good, but it calls the backend API incorrectly when it tries to update the user selected image which results in the `object` being stored as an image name instead of actual `image:tag` value

**Comment by jeff-phillips-18** [comment]
@mroman-redhat @vpavlin Fixed the call to set the selected image

@kywalker-rh Updated the layout so that only the expanded item takes up space. Added the notebook description to the popover.

image

**Comment by vpavlin** [comment]
Ok, it now updates the user CM, but it seems the value is still incorrect - if the is only 1 tag, the value in CM is `image:`, not `image:tag`

```
last_selected_image: 's2i-generic-data-science-notebook:'
```

vs. 

```
last_selected_image: 's2i-generic-data-science-notebook:v0.0.4'
```

Also, I noticed that when there is multiple tags there is difference in what you click on:

1. Clicking on the image name results in the problem above
2. Clicking on the tag name results in a correct image name being used

One last thing - when I select Minimal, version 0.0.14 a word `(default)` shows up next to the `Minimal Python v 0.0.14`
Screenshot from 2021-07-01 14-19-30
Screenshot from 2021-07-01 14-19-25

**Comment by jeff-phillips-18** [comment]
Thanks @vpavlin. Fixed.

**Review by mroman-redhat — APPROVED** [review]
Tested and the image versions seem to be rendered correctly and written to the user configmap correctly as well.

**Pull request merged**